woocommerce/grow
# Upgrade JS packages `jsdoc` and `tracking-jsdoc` to use Node.js v20

### Changes proposed in this Pull Request:

Part of #108

This PR:

- Use Node.js v20 to regenerate the package-lock.json of JS package `tracking-jsdoc`.
- Set minimum support for Node.js to v16.
- Add the 'node:' prefix to explicitly indicate it requires a Node.js core module.
- Avoid generating package-lock.json file after `tracking-jsdoc` installation when developing this repo.

💡 The `<repo_directory>/.nvmrc` will be updated in a separate PR.

### Detailed test instructions:

1. Use Node.js v20
2. `cd packages/js/jsdoc`
3. `npm install`
4. `npm run dev:doc:tracking`
5. Check if the generated content in `<repo_directory>/samples/jsdoc/TRACKING.md` is the same
6. Repeat the test with Node.js v16
